consensus/src/dag/dag_driver.rs

```diff
@@ -290,6 +286,36 @@ impl DagDriver {
             prev_handle.abort();
         }
     }
+
+    fn calculate_payload_dimensions(&self, round: Round) -> (u64, u64) {
+        let (max_txns_per_round, max_size_per_round_bytes) =
+            if let Some((backoff_max_txns, backoff_max_size_bytes)) =
+                self.chain_backoff.get_round_payload_limits(round)
+            {
+                (
+                    self.payload_config
+                        .max_sending_txns_per_round
+                        .min(backoff_max_txns),
+                    self.payload_config
+                        .max_sending_size_per_round_bytes
+                        .min(backoff_max_size_bytes),
+                )
+            } else {
+                (
+                    self.payload_config.max_sending_txns_per_round,
+                    self.payload_config.max_sending_size_per_round_bytes,
+                )
+            };
+        // TODO: warn/panic if division yields 0 txns
+        let max_txns = max_txns_per_round
+            .saturating_div(self.epoch_state.verifier.len() as u64)
```

Comment:
This is not accurate. If we are here then we have validators behind. So to be accurate we need to divide by the number of healthy validators.